### ISSUE TITLE
gradient_accumulation with deepspeed 

### ISSUE THREAD

**Issue by yoctta** [huggingface/accelerate#1767]
### System Info

```Shell
- `Accelerate` version: 0.21.0
- Platform: Linux-3.10.0-957.el7.x86_64-x86_64-with-glibc2.17
- Python version: 3.10.12
- Numpy version: 1.25.0
- PyTorch version (GPU?): 2.0.1 (False)
- PyTorch XPU available: False
- PyTorch NPU available: False
- System RAM: 723.61 GB
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: DEEPSPEED
        - mixed_precision: bf16
        - use_cpu: False
        - num_processes: 8
        - machine_rank: 0
        - num_machines: 1
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - deepspeed_config: {'gradient_accumulation_steps': 16, 'gradient_clipping': 3.0, 'offload_optimizer_device': 'none', 'offload_param_device': 'none', 'zero3_init_flag': False, 'zero_stage': 2}
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []
        - dynamo_config: {'dynamo_backend': 'INDUCTOR', 'dynamo_mode': 'reduce-overhead', 'dynamo_use_dynamic': False, 'dynamo_use_fullgraph': False}
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

examples/by_feature/deepspeed_with_config_support.py

Line 643             loss = loss / args.gradient_accumulation_steps

~~~~~~~~~~~~~~~~~~~~~

https://github.com/huggingface/accelerate/blob/0b36ca6e64245b123e9455521c1bd985f4b72679/src/accelerate/accelerator.py#L1901

        if self.distributed_type != DistributedType.DEEPSPEED:
            # deepspeed handles loss scaling by gradient_accumulation_steps in its `backward`
            loss = loss / self.gradient_accumulation_steps
        if self.distributed_type == DistributedType.DEEPSPEED:
            self.deepspeed_engine_wrapped.backward(loss, **kwargs)

### Expected behavior

Do I need to MANUALLY div loss by gradient_accumulation_steps when using deepspeed config?

**Comment by sgugger** [comment]
cc @pacman100

**Comment by pacman100** [comment]
Hello, it is a mistake in the DeepSpeed example. I will raise a PR to fix this. Thank you for the pointers.

**Comment by pacman100** [comment]
DeepSpeed handles Gradient accumulation on its own, no need to do anything in your script.

**Comment by pacman100** [comment]
above PR should resolve this
